internal/pkg/cli/job_package.go

```diff
@@ -0,0 +1,180 @@
+)
+
+const (
+	jobPackageJobNamePrompt = "Which job would you like to generate a CloudFormation template for?"
+	jobPackageEnvNamePrompt = "Which environment would you like to package this stack for?"
+)
+
+type packageJobVars struct {
+	name      string
+	envName   string
+	appName   string
+	tag       string
+	outputDir string
+}
+
+type packageJobOpts struct {
+	packageJobVars
+
+	// Interfaces to interact with dependencies.
+	ws     wsJobDirReader
+	store  store
+	runner runner
+	sel    wsSelector
+	prompt prompter
+}
+
+func newPackageJobOpts(vars packageJobVars) (*packageJobOpts, error) {
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, fmt.Errorf("new workspace: %w", err)
+	}
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("connect to config store: %w", err)
+	}
+	prompter := prompt.New()
+	opts := &packageJobOpts{
+		packageJobVars: vars,
+		ws:             ws,
+		store:          store,
+		runner:         command.New(),
+		sel:            selector.NewWorkspaceSelect(prompter, store, ws),
+		prompt:         prompter,
+	}
+	return opts, nil
+}
+
+// Validate returns an error if the values provided by the user are invalid.
+func (o *packageJobOpts) Validate() error {
+	if o.appName == "" {
+		return errNoAppInWorkspace
+	}
+	if o.name != "" {
+		names, err := o.ws.JobNames()
+		if err != nil {
+			return fmt.Errorf("list jobs in the workspace: %w", err)
+		}
+		if !contains(o.name, names) {
+			return fmt.Errorf("job '%s' does not exist in the workspace", o.name)
+		}
+	}
+	if o.envName != "" {
+		if _, err := o.store.GetEnvironment(o.appName, o.envName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Ask prompts the user for any missing required fields.
+func (o *packageJobOpts) Ask() error {
+	if err := o.askAppName(); err != nil {
+		return err
+	}
+	if err := o.askEnvName(); err != nil {
+		return err
+	}
+	return o.askTag()
+}
+
+// Execute prints the CloudFormation template of the application for the environment.
+func (o *packageJobOpts) Execute() error {
+	return nil
+}
+
+func (o *packageJobOpts) askAppName() error {
+	if o.name != "" {
+		return nil
+	}
+
+	name, err := o.sel.Job(jobPackageJobNamePrompt, "")
+	if err != nil {
+		return fmt.Errorf("select job: %w", err)
+	}
+	o.name = name
+	return nil
+}
+
+func (o *packageJobOpts) askEnvName() error {
+	if o.envName != "" {
+		return nil
+	}
+
+	name, err := o.sel.Environment(jobPackageEnvNamePrompt, "", o.appName)
+	if err != nil {
+		return fmt.Errorf("select environment: %w", err)
+	}
+	o.envName = name
+	return nil
+}
+
+func (o *packageJobOpts) askTag() error {
+	if o.tag != "" {
+		return nil
+	}
+
+	tag, err := getVersionTag(o.runner)
+	if err != nil {
+		// We're not in a Git repository, prompt the user for an explicit tag.
+		tag, err = o.prompt.Get(inputImageTagPrompt, "", prompt.RequireNonEmpty)
+		if err != nil {
+			return fmt.Errorf("prompt get image tag: %w", err)
+		}
+	}
+	o.tag = tag
```

Comment:
can we create a separate function in `git.go` so that both `job package` and `svc package` re-use this piece?

```go
func askImageTag(prompt prompter, cmd runner) (string, error) {
  ...
}
```

Reply:
Should `svc_deploy` and `job_deploy` use it as well? They are very similar except for `log.Warningln("Failed to default tag, are you in a git repository?")` between `getVersionTag` and the prompt.

Reply:
Sounds goooooood! we can add the log warning so that it also appears in package, I think  it's fine :)

Reply:
As discussed offline, added the custom log warning and added Stderr to `command.go` so the "fatal" error doesn't appear in addition to the log warning.